src/lib_db.ml

```diff
@@ -115,19 +117,24 @@ let create findlib ~scopes internal_libraries =
     { findlib
     ; by_public_name   = Hashtbl.create 1024
     ; by_internal_name = Hashtbl.create 1024
+    ; scope_by_name    = Hashtbl.create 1024
     ; instalable_internal_libs = String_map.empty
     ; local_public_libs
     }
   in
   (* Initializes the scopes, including [Path.root] so that when there are no <pkg>.opam
      files in parent directories, the scope is the whole workspace. *)
+  let root_scope = { libs = String_map.empty; scope = Scope.empty } in
+  Hashtbl.add t.by_internal_name ~key:Path.root ~data:root_scope;
+  Hashtbl.add t.scope_by_name ~key:"" ~data:root_scope.scope;
   List.iter scopes ~f:(fun (scope : Scope.t) ->
     Hashtbl.add t.by_internal_name ~key:scope.root
       ~data:{ libs = String_map.empty
             ; scope
             });
   List.iter internal_libraries ~f:(fun ((dir, lib) as internal) ->
     let scope = internal_name_scope t ~dir in
+    Hashtbl.add t.scope_by_name ~key:scope.scope.name ~data:scope.scope;
```

Comment:
It doesn't feel natural to do this here. If there are no libs in the scope, the entry won't be initialized, and if there are multiple, it will be initialized several times. We should iter over the list of scopes instead.

Reply:
Yeah, that was an artifact from when I was inserting the paths. Good catch.